crates/exchange-rate-oracle/src/lib.rs

```diff
@@ -259,30 +243,82 @@ pub mod pallet {
 
 #[cfg_attr(test, mockable)]
 impl<T: Config> Pallet<T> {
-    fn begin_block(_height: T::BlockNumber) {
-        if Self::is_max_delay_passed() {
-            Self::report_oracle_offline();
+    // public only for testing purposes
+    pub fn begin_block(_height: T::BlockNumber) {
+        // read to a temporary value, because we can't alter the map while we iterate over it
+        let raw_values_updated: Vec<_> = RawValuesUpdated::<T>::iter().collect();
```

Comment:
You had mentioned that orml-oracle limits exchange-rate submission to one per block. This might prove useful here as a spam prevention mechanism.

Reply:
Well currently oracle submission is fully paid for, so we have spam protection. In the future, it might be nice to have a more strict spam protection rule: we could specify that oracles are free to make submissions if either (1) more than x amount of time passed since its last submission, or (2) the exchange rate changed by y%

I would prefer to leave this as a future improvement/optimization though